### DISCUSSION TITLE
Token Exchange with Istio

### DISCUSSION THREAD

**Discussion by Steffi14** [istio/istio#49996] [Q&A]
Hi,

I try to find a solution for a token exchange with Istio. Is there anything already available?

So in detail: If a request comes with a TPP Token, I would like to have the authentication policy to check the token against a public key and then get a new token from our internal keycloak, as the services can only work with the internal token. 
The problem ist that I am not allowed to use the token exchange feature directly from keycloak as it is only a preview feature nor I am allowed to write a wasm plugin as it is alpha feature.

Is there anyone who knows how to get to a solution with istio?

**Comment by PT-GD** [comment]
Did you make any progress on this? Token Exchange v2 is a thing. I have basically the same problem (just asked the question in #security on Istio's Slack)
